docs/middleware/README.md

```diff
@@ -0,0 +1,86 @@
+# AVS Smart Contract Srchitecture
+
+## Introduction
+
+EigenLayer AVSs are a new type of protocol that make use of EigenLayer’s restaking primitive. AVSs are different from current chains and other smart contract protocols in that they are validated by EigenLayer operators. There are are 3 specific types of conditions that AVSs implement in smart contracts onchain:
+- Registration/Deregistration conditions: What requirements do operators need to have in order to register for/deregister from the AVS?
+- Payment conditions: How much does a certain operator deserve to be paid for their validation? In what form are they paid?
+- Slashing conditions: What behavior is not allowed of operators by the AVS? What exact mechanism should be used to determine this behavior onchain?
+
+The EigenLabs dev team has been building out a smart contract architecture for AVSs to provide a base for AVSs to build on top of for their specific use case. They have been using it internally for the first AVS on EigenLayer: EigenDA.
+
+## The Registry Coordinator and Registries 
+
+![Registry Architecture](../images/registry_architecture.png)
```

Comment:
It might be on my end, but I am having an issue rendering this link in the markdown

Reply:
Good catch, should work now: 36d78f3f